### ISSUE TITLE
Radio buttons are not linked to their `label`s

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2432]
Radio button labels are not matching the radio inputs - see screenshot for details:

![image](https://github.com/trimble-oss/modus-web-components/assets/1212885/4a1211c6-f9c9-406d-b03a-6d317ec7b633)

Radio should have something like: `id="mwc_id_0_radio_input"` and then the `label` should have `for="mwc_id_0_radio_input"`.
